src/Misc.ino

```diff
@@ -2629,7 +2629,7 @@ void rulesTimers()
   {
     if (!RulesTimer[x].paused && RulesTimer[x].timestamp != 0L) // timer active?
     {
-      if (timeOutReached(RulesTimer[x].timestamp)) // timer finished?
+      if (timeOutReached(RulesTimer[x].timestamp - 100)) // timer finished?  (minus 100 ms)
```

Comment:
You should not subtract from a time.
The time will wrap (unsigned int), but subtracting from that will result in strange results.
This will be after either 47.9 days, or half that period.

You could use `timediff`, which returns a signed int, which can be compared.

Reply:
This will be ok ?

   -- if (timeOutReached(RulesTimer[x].timestamp - 100)) // timer finished?  (minus 100 ms)

  ++ if (timeDiff(millis(),RulesTimer[x].timestamp)  < 100)

Ps. I decided for 100 ms because  runOncePerSecond()  is not quite quite accurate and repeatable
as it should

Reply:
I always have to check in the code to see the definition of timeDiff, so perhaps there should be a function called `timeUntil`.

But that's the general idea indeed.
And if rules timers have to be more accurate, we should add something to do just that, instead of applying some average correction offset. Maybe the suggested cron timer approach suggested in another issue would be a good addition for this.
Or we should add some timer to be set in the rules at which time the rules should be processed.
I want to fix it in such a way that it is beneficial for more use cases.
And applying a fixed offset to correct some timing issue is almost always not the best solution.

Reply:
Globally, this is not important, but when I set the timer for 1 second I had 2 seconds, when I set it for 2 seconds, I had 3 seconds (100% and 50% error). With longer times the error is acceptable

Reply:
ok, I close this pull